### PR TITLE
Replace requirejs-rails with webpacker

### PR DESCRIPTION
**Pull request by monfresh**
**Why**: requirejs-rails is not very well maintained, and is preventing this app from being upgraded to Rails 5. Webpacker is the new recommended way of handling JS.

### CODE REVIEW

**Comment by monfresh** [comment]
@anselmbradford I'd love your eyes on this if you get a chance!

**Comment by anselmbradford** [comment]
I'll try this weekend! Ping me if I forget

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      "useBuiltIns": true
+    }]
+  ],
+
+  "plugins": [
+    "syntax-dynamic-import",
+    "transform-object-rest-spread",
+    ["transform-class-properties", { "spec": true }]
+  ]
+}
```
Comment (monfresh) [inline .babelrc]: This file was automatically generated after running `bundle exec rake webpacker:install`

File: .gitignore [hunk 2]
```diff
@@ -53,3 +53,8 @@ rerun.txt
 /config/application.yml
 
 .gemnasium.yml
+/public/packs
+/public/packs-test
+/node_modules
+yarn-debug.log*
+.yarn-integrity
```
Comment (monfresh) [inline .gitignore]: Ther

**Comment by monfresh** [comment]
@anselmbradford To give you more context, for the most part what I did here was convert the JS from AMD to ES6 syntax using a converter tool I found. I don't remember which one exactly. Maybe [this one](https://www.npmjs.com/package/@buxlabs/amd-to-es6) or [this one](https://github.com/jonbretman/amd-to-as6). Overall, the tool worked great. All it did really was replace the `require` and `define` calls with `import`, and replace the final `return` with an `export`. It doesn't convert `var` to `let` or `const` or do other ES6 conversions.

Once that was done, I didn't make any other changes to the JS except for these files:

- I removed the error handling in `result-init`, `detail-init`, and `detail-map` because I wasn't sure how to implement it in ES6, and I didn't think it was necessary anymore because it looked like it was related to handling RequireJS modules failing to load.

- I removed the async loading of google maps in `map-loader` because I couldn't get it to work with `OverlappingMarkerSpiderfier`, which kept complaining that `google` wasn't defined. Instead, what I did was load all the application JS after the main content. That way, the site can still be used while maps are loading. I tested the site locally and it seemed to work fine. This is not my area of expertise, so any suggestions you can provide would be welcome!

**Comment by monfresh** [comment]
I'm going to update the PR to make it easier to see the diff between the old and new JS files. I think the way the tool converted the files made git think the old ones were deleted instead of renamed.

**Comment by monfresh** [comment]
I also deleted all of the files in the `remy` folder except for `classList.js` because it looked like that was the only one being used. Do you remember if any other ones were being used?

I also deleted domReady because it looks like that is specific to RequireJS and no longer needed. Is that correct?

**Comment by monfresh** [comment]
Closing in favor of #868

**Pull request closed without merging**